identity/src/main/java/com/android/identity/SessionEncryptionReader.java

```diff
@@ -191,6 +191,59 @@ void setSendSessionEstablishment(boolean sendSessionEstablishment) {
         return messageData;
     }
 
+    // Only used for testing, will produce a SessionEstablishment message with an
+    // invalid COSE_Key for EReaderKeyq
+    @NonNull byte[] encryptMessageToDeviceWithInvalidEReaderKey(
```

Comment:
This function is the same as `encryptMessageToDevice` with only `DataItem eReaderKey = Util.cborBuildCoseKeyWithMalformedYPoint(mEReaderKeyPublic);` changed. While it makes sense to copy code into a new function in order to avoid including a boolean as a flag for malformations in the public `encryptMessageToDevice` function, an alternate solution could be to wrap all this code in a function like `encryptMessageToDeviceHelper(@Nullable byte[] messagePlaintext, @NonNull OptionalLong statusCode, boolean malformed)` and then make the two functions be the following:

```
encryptMessageToDevice(@Nullable byte[] messagePlaintext, @NonNull OptionalLong statusCode) {
     encryptMessageToDeviceHelper(messagePlaintext, statusCode, false); }

encryptMessageToDeviceWithInvalidEReaderKey(@Nullable byte[] messagePlaintext, @NonNull OptionalLong statusCode) {
     encryptMessageToDeviceHelper(messagePlaintext, statusCode, true); }
```